spec/sync/storage-manager-listener-spec.js

```diff
@@ -0,0 +1,43 @@
+const getStorageManagerListener = require('./../../lib/sync/storage-manager-listener');
+
+describe('storageManagerListener', () => {
+  it('returns a function', () => {
+    expect(typeof getStorageManagerListener).toEqual('function');
+  });
+
+  describe('when a store is not passed in the first time it is invoked', () => {
+    it('throws an error', () => {
+      expect(() => getStorageManagerListener()).toThrow(new Error('You must pass in a redux store the first time you call getStorageManagerListener'));
+      expect(() => getStorageManagerListener({})).toThrow(new Error('You must pass in a redux store the first time you call getStorageManagerListener'));
+    });
+  });
+
+  describe('when a store is passed in the first time it is invoked', () => {
+    it('returns the same function every time it is invoked after that', () => {
+      const firstReturn = getStorageManagerListener({ dispatch: () => {} });
+      expect(typeof firstReturn).toEqual('function');
+      expect(getStorageManagerListener()).toBe(firstReturn);
+      expect(getStorageManagerListener()).toBe(firstReturn);
+      expect(getStorageManagerListener()).toBe(firstReturn);
+    });
+  });
+
+  describe('when a store is passed in again', () => {
+    describe('when it is the same store', () => {
+      it('returns the same listener', () => {
+        const store = { dispatch: () => {} };
+        const listenerWithFirstStore = getStorageManagerListener(store);
+        const listenerWithOtherStore = getStorageManagerListener(store);
+        expect(listenerWithFirstStore).toBe(listenerWithOtherStore);
+      });
+    });
+
+    describe('when it is a different store', () => {
+      it('overrides the original store and returns a new handler bound to the new store', () => {
```

Comment:
To follow up my related comment above, if we want this behavior shouldn't we clean up the event handlers from the previous store first?

Still curious what the use case for this case would be.